response.go

```diff
@@ -179,34 +160,228 @@ func (c *CountResultItem) String() string {
 	return fmt.Sprintf("%d:%d", c.ID, c.Count)
 }
 
-func countItemsFromInternal(items []*pbuf.Pair) []*CountResultItem {
+func countItemsFromInternal(items []*pbuf.Pair) TopNResult {
 	result := make([]*CountResultItem, 0, len(items))
 	for _, v := range items {
 		result = append(result, &CountResultItem{ID: v.ID, Key: v.Key, Count: v.Count})
 	}
-	return result
+	return TopNResult(result)
+}
+
+type TopNResult []*CountResultItem
+
+func (TopNResult) Type() uint32 {
+	return QueryResultTypePairs
+}
+
+func (TopNResult) Bitmap() BitmapResult {
+	return defaultBitmapResult()
+}
+
+func (t TopNResult) CountItems() []*CountResultItem {
+	return t
+}
+
+func (TopNResult) Count() int64 {
+	return 0
+}
+
+func (TopNResult) Sum() int64 {
+	return 0
+}
+
+func (TopNResult) Changed() bool {
+	return false
 }
 
 // BitmapResult represents a result from Bitmap, Union, Intersect, Difference and Range PQL calls.
 type BitmapResult struct {
-	Attributes map[string]interface{}
-	Bits       []uint64
-	Keys       []string
+	Attributes map[string]interface{} `json:"attrs"`
+	Bits       []uint64               `json:"bits"`
+	Keys       []string               `json:"keys"`
 }
 
 func newBitmapResultFromInternal(bitmap *pbuf.Bitmap) (*BitmapResult, error) {
 	attrs, err := convertInternalAttrsToMap(bitmap.Attrs)
 	if err != nil {
 		return nil, err
 	}
+	bits := bitmap.Bits
+	if bits == nil {
+		bits = []uint64{}
+	}
+	keys := bitmap.Keys
+	if keys == nil {
+		keys = []string{}
+	}
 	result := &BitmapResult{
 		Attributes: attrs,
-		Bits:       bitmap.Bits,
-		Keys:       bitmap.Keys,
+		Bits:       bits,
+		Keys:       keys,
 	}
 	return result, nil
 }
 
+func defaultBitmapResult() BitmapResult {
```

Comment:
BitmapResult{} should be the default IMO. Accessing a nil map like `a, ok := map["blah"]` does not panic. Iterating over a nil map does not panic.